src/rapids_pre_commit_hooks/pyproject_license.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+import uuid
+
+import tomlkit
+import tomlkit.exceptions
+
+from .lint import LintMain
+
+RAPIDS_LICENSE = "Apache 2.0"
+ACCEPTABLE_LICENSES = {
+    "Apache 2.0",
+    "BSD-3-Clause",
+}
+
+
+def find_value_location(document, key):
+    copied_document = copy.deepcopy(document)
+    placeholder = uuid.uuid4()
+
+    node = copied_document
+    while len(key) > 1:
+        node = node[key[0]]
+        key = key[1:]
+    old_value = node[key[0]]
+    node[key[0]] = str(placeholder)
+
+    begin_loc = copied_document.as_string().find(
+        tomlkit.string(str(placeholder)).as_string()
+    )
+    end_loc = begin_loc + len(old_value.as_string())
+    return begin_loc, end_loc
+
+
+def check_pyproject_license(linter, args):
+    document = tomlkit.loads(linter.content)
+    try:
```

Comment:
There's some duplication here that we could clean up. What if instead of try-excepts, we check for the project key, and then the license key, and in each case set a flag `key_missing` or so and combine that with `project_table.is_super_table` to indicate that we insert the value from scratch?